src/confluent_kafka/src/Admin.c

```diff
@@ -1531,6 +1528,361 @@ const char Admin_list_consumer_groups_doc[] = PyDoc_STR(
+                        PyErr_Format(PyExc_TypeError,
+                                     "Alteration %d: user field should be a string, got %s",
+                                     i,
+                                     ((PyTypeObject *)PyObject_Type(user))->
+                                     tp_name);
+                        goto err;
+                }
+
+                Py_DECREF(user);
+                user_c = cfl_PyUnistr_AsUTF8(u_user, &uo_user);
+
+                if(PyObject_IsInstance(alteration,UserScramCredentialUpsertion_type)){
+                        /* Upsertion Type*/
+                        cfl_PyObject_GetAttr(alteration,"salt",&salt,NULL,0,0);
+                        if (salt != Py_None && Py_TYPE(salt) != &PyBytes_Type) {
+                                PyErr_Format(PyExc_TypeError,
+                                        "Alteration %d: salt field should be bytes, got %s",
+                                        i,
+                                        ((PyTypeObject *)PyObject_Type(salt))->
+                                        tp_name);
+                                goto err;
+                        }
+                        if (salt != Py_None) {
+                                PyBytes_AsStringAndSize(salt, (char **) &salt_c, &salt_c_size);
+                        }
+
+                        cfl_PyObject_GetAttr(alteration,"password",&password,NULL,0,0);
+                        if (Py_TYPE(password) != &PyBytes_Type) {
+                                PyErr_Format(PyExc_TypeError,
+                                        "Alteration %d: password field should be bytes, got %s",
+                                        i,
+                                        ((PyTypeObject *)PyObject_Type(password))->
+                                        tp_name);
+                                goto err;
+                        }
+                        PyBytes_AsStringAndSize(password, (char **) &password_c, &password_c_size);
+
+                        cfl_PyObject_GetAttr(alteration,"scram_credential_info",&scram_credential_info,NULL,0,0);
+                        if (!PyObject_IsInstance(scram_credential_info, ScramCredentialInfo_type)) {
+                                PyErr_SetString(PyExc_TypeError,
+                                        "scram_credential_info should be of ScramCredentialInfo type");
+                                goto err;
+                        }
+
+                        cfl_PyObject_GetInt(scram_credential_info,"iterations",&iterations,0,1);
+                        cfl_PyObject_GetAttr(scram_credential_info,"mechanism",&mechanism,NULL,0,0);
+                        if (!PyObject_IsInstance(mechanism, ScramMechanism_type)) {
+                                PyErr_SetString(PyExc_TypeError,
+                                        "mechanism should be of ScramMechanism type");
+                                goto err;
+                        }
```

Comment:
We can move this out as this is mandatory in both kind of alteration.

Reply:
Here it's inside ScramCredentialInfo while in deletion is directly in the alteration without iterations